File: Aegis-Nexus/backend/models/service.py
from dataclasses import dataclass, field
from typing import Literal

ServiceStatus = Literal["Healthy", "Warning", "Critical"]
ServiceKind = Literal["frontend", "api", "worker", "database"]


@dataclass(slots=True)
class Dependency:
    id: str
    name: str
    status: ServiceStatus = "Healthy"


@dataclass(slots=True)
class Service:
    id: str
    name: str
    kind: ServiceKind
    description: str
    owner: str
    dependencies: list[Dependency] = field(default_factory=list)
    status: ServiceStatus = "Healthy"
    available: bool = True

File: Aegis-Nexus/frontend/types/service.ts
export type ServiceStatus = "Healthy" | "Warning" | "Critical";

export interface Dependency {
  id: string;
  name: string;
  status: ServiceStatus;
}

export interface MonitoredService {
  id: string;
  name: string;
  kind: "frontend" | "api" | "worker" | "database";
  description: string;
  owner: string;
  status: ServiceStatus;
  available: boolean;
  dependencies: Dependency[];
}
